analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -266,30 +313,32 @@ def get_checker_list(self, config) -> Tuple[List[str], List[str]]:
         clang-tidy and in some cases that would cause analysis error due to
         some ClangSA bug.
         """
-        checkers = []
         # Usage of a set will remove compiler warnings and clang-diagnostics
         # which are the same.
+        # In clang-tidy the warnings as compiler-diagnostics are enabled and
+        # disabled by -W<name> and -Wno-<name> respectively. The
+        # clang-diagnostic-<name> is needed, because by default all checkers
```

Comment:
The part after "because" doesn't explain to me why we can't preserve `-W` and -`Wno`. It just runs in a circle.

Reply:
> Would you think, it explains?

No.

Reply:
I think it explains. We do preserve `-W` and `-Wno`, but it's not enough, because we disable every diagnostics by default.
```
clang-tidy -checks=-* main.cpp -- -W<some_warning>
```
This command doesn't emit any diagnostics, because `-checks=-*` disables it. That's why we need to add `clang-diagnostic-<some_warning>` too:
```
clang-tidy -checks=-*,clang-diagnostic-<some_warning> main.cpp -- -W<some_warning>
```
If the phrasing is not clear, then I'd agree to rewrite it.